### PR TITLE
Deregistration UI websocket events

### PR DESCRIPTION
**Pull request by rtorrero**
# Description
This PR adds the code to handle the state changes in react upon the arrival of the `*_deregistered` events from the host, cluster and sap_system aggregates.

Remember to add the right labels to this PR.

## How was this tested?
Manually, issuing Host deregistration requests in the REPL. End-to-end tests will be added once the UI changes are completed.

### CODE REVIEW

**Review by arbulu89 — COMMENTED** [review]
@rtorrero I think this is a good chance to start adding jest unit tests to this changes

File: assets/js/state/clusters.test.js
Comment (arbulu89) [inline assets/js/state/clusters.test.js:5]: Most probably it would be better to use the `clusterFactory`.
```
const { id } = clusterFactory.build()
# or
const clusters = cluster.Factory.buildList(5) # with some further work, which would do a more robust test, just removing 1 from the list, instead of looking for an empty list
```

The same goes for the other tests

File: assets/js/state/sapSystems.js
Comment (arbulu89) [inline assets/js/state/sapSystems.js:105]: Following the others you would do `sapSystem.id !== id`

File: assets/js/state/sagas/clusters.js
Comment (arbulu89) [inline assets/js/state/sagas/clusters.js:16]: Keep in mind that not all clusters have a name, so this could end up empty.
Maybe something like `${payload.name || payload.id}` ?

File: assets/js/state/sagas/clusters.test.js
Comment (arbulu89) [inline assets/js/state/sagas/clusters.test.js:6]: Maybe `should remove the cluster` simply?

File: assets/js/state/sagas/clusters.test.js
Comment (arbulu89) [inline assets/js/state/sagas/clusters.test.js:7]: Use factories

File: assets/js/state/sagas/index.js
Comment (arbulu89) [inline assets/js/state/sagas/index.js:212]: Let's move this `watch` to the saga files (as we do with the catalog and lastExecutions).
Besides that, `HOST_DEREGISTERED` should be a constant.
By now, these are stored in the `actions` file, but with the latest ADR i'm not sure where this should go. Reducer or saga? @dottorblaster wdyt?

File: lib/trento/application/projectors/sap_system_projector.ex
Comment (arbulu89) [inline lib/trento/application/projectors/sap_system_projector.ex:212]: Doesn't this raise some any failure in tests?
We should have a look, or at least try to include this change in the tests. 
Maybe the assertion looks for partial objects

**Review by arbulu89 — CHANGES_REQUESTED** [review]
In general it looks good.
I think that we should use more the factories in the tests if possible.
Besides that, I commented some topics about the placement of functions and usage of constants.

Finally, What about the `databases`?

File: assets/js/state/sagas/clusters.test.js
Comment (jamie-suse) [reply assets/js/state/sagas/clusters.test.js:6]: This test, is testing that the saga dispatches the correct action to the reducer. i.e. it does not validate the state change (because we test this in the reducer test).

Comment (arbulu89) [reply assets/js/state/sagas/clusters.test.js:8]: But that's going technical...
At the end this saga aims to update our state internally, in our domain terminology this is to `remove a cluster`.
Anyway, I don't want to force anything, it is just my point of view

Comment (jamie-suse) [reply assets/js/state/sagas/clusters.test.js]: Okay no worries, changed it. Was just making sure we are on the same page 👍

File: assets/js/state/clusters.test.js
Comment (arbulu89) [inline assets/js/state/clusters.test.js:14]: I still think that this test is weak. I would prefer having a removal from a list where you have more than one element

**Review by arbulu89 — COMMENTED** [review]
Hey @jamie-suse 
It looks much better.
Waiting to confirm where to put the new `action` constants.
And please, have a look on the comments I wrote about the `id/cluster_id` possible difference.
I think we would be fine to merge once that is checked

File: assets/js/state/actions/cluster.js
Comment (arbulu89) [inline assets/js/state/actions/cluster.js:1]: I don't understand why you do these changes in this PR :sweat_smile: 
I would create tech-debt tasks and work on them. 
Otherwise the reviews got out of hand.
(And I know that I asked you to not use action files, but you could have only do things belonging to the actual change there)

File: assets/js/state/sagas/clusters.js
Comment (arbulu89) [inline assets/js/state/sagas/clusters.js:7]: It was not simply better to change the projection code to use `id` instead of `cluster_id`?
To avoid transformations that we might have issues finding in the future

File: assets/js/state/sagas/hosts.test.js
Comment (arbulu89) [inline assets/js/state/sagas/hosts.test.js:8]: This doesn't make much sense, right?
Why not simply do `const payload = hostFactory.build()` ?
Having more fields from the factory doesn't affect anything as far as I see

**Review by arbulu89 — APPROVED** [review]
@jamie-suse I'm approving this because I see it is stalling.
But I still think that there are things to be improved.

Other thing that now I miss is the instances live update.
We don't listen to any message to remove the sap app/database instances, so what happens when we remove them?
I would expect that they are removed from many views (host, sap system, database).
Could you track this missing thing and still work on it?

File: assets/js/state/sagas/hosts.test.js
Comment (jamie-suse) [reply assets/js/state/sagas/hosts.test.js:8]: The extra fields cause the check using `toContainEqual()` to fail. Trying to see if there's an easy way to get it to work in Jest.

File: assets/js/state/sagas/clusters.test.js
Comment (CDimonaco) [inline assets/js/state/sagas/clusters.test.js:11]: We should use the value of CLUSTER_DEREGISTERED as expected value, not the const imported,

File: assets/js/state/sagas/hosts.test.js
Comment (CDimonaco) [inline assets/js/state/sagas/hosts.test.js:12]: Why we need the type field here>
I think we don't need the type value here, we could just easily add the payload like the other tests

**Review by CDimonaco — CHANGES_REQUESTED** [review]

File: assets/js/state/sagas/clusters.js
Comment (jamie-suse) [reply assets/js/state/sagas/clusters.js:9]: Okay, will do 👍
Also other sagas in the codebase use `appendEntryToLiveFeed()` though, shouldn't we be removing them also? I can create a tech-debt ticket for this

**Comment by stefanotorresi** [comment]
@jamie-suse I understood this was still undergoing review, why did you self-merge it?

**Pull request merged**
